lib/open_api/serializer.rb

```diff
@@ -1,14 +1,43 @@
 module OpenApi
   module Serializer
     def as_json(arg = {})
-      return super unless kind_of?(ApplicationRecord)
+      previous = super
+      encrypted_columns_set = (self.class.try(:encrypted_columns) || []).to_set
+      encryption_filtered = previous.except(*encrypted_columns_set)
+      version = version_from_path(arg)
 
-      super.tap do |hash|
-        hash.each_key do |attr_key|
-          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
-          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
-        end
+      return encryption_filtered unless arg.key?(:prefixes) || version
+
+      # We use "#{klass.name}Out" for the definitions on what are returned.
+      schema = Api::Docs[version].definitions[self.class.name + "Out"]
```

Comment:
how does it work if `version.nil? && arg.key?(:prefixes)`?

Reply:
If version is nil then everything stops.  We could add a raise to help troubleshoot why `version` is nil, but since it has to do with the request, if version doesn't return than nothing can move forward with the serializer.